test/003_staking_rewards_test.js

```diff
@@ -126,6 +126,65 @@ describe("Staking Rewards Contract", function () {
         let updatedRewardPerToken = ethers.BigNumber.from(BigNumber.from(rewardRate).mul(timestamp - lastUpdateTime).mul(multiplier).div(totalSupply))
         expect(await stakingRewards.rewardPerToken()).to.be.eq(updatedRewardPerToken)
     });
+
+    it('Can update max reward', async function () {
+        rewardRateBeforeUpdateMaxReward = rewardRate;
+        await increaseTime(4 * DAY)
+        await expect(stakingRewards.connect(accounts[2]).updateMaxReward(BigNumber.from(10).pow(21))).to.be.reverted;
+        const updateMaxRewardTxn = await stakingRewards.updateMaxReward(BigNumber.from(10).pow(35));
+        // console.log("updateMaxRewardTxn: ", updateMaxRewardTxn)
+        const updateMaxRewardReceipt = await updateMaxRewardTxn.wait();
+        // console.log("waiting for txn to mine / receipt : ", updateMaxRewardReceipt);
+        await expect(updateMaxRewardTxn).to.emit(stakingRewards, "RewardRateUpdated");
+        // 
+        // postMaxrewardsUpdatedTimestamp = (await ethers.provider.getBlock('latest')).timestamp
+        postMaxrewardsUpdatedTimestamp = (await ethers.provider.getBlock(updateMaxRewardReceipt.blockNumber)).timestamp
+        expect(await stakingRewards.maxReward()).to.eq(BigNumber.from(10).pow(35))
+        rewardRateAfterUpdateMaxReward = await stakingRewards.rewardRate();
+    });
+
+    it("Rewards to be generated over contract lifetime should be less than max rewards",async ()=>{
+        // t=1 deployment time
```

Comment:
@ShubhamChndrvnshi Looks good to merge, if we can fix these comments and what we discussed in standup.

Reply:
incorporated them in latest commit